app/assets/javascripts/geolocation.js.erb

```diff
@@ -0,0 +1,31 @@
+document.addEventListener('DOMContentLoaded', function() {
+  document.getElementById('useLocation').addEventListener('click', function() {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(function(position) {
+        const latitude = position.coords.latitude;
+        const longitude = position.coords.longitude;
+
+        const url = `/geocoder/city_from_location?latitude=${latitude}&longitude=${longitude}`;
```

Comment:
This looks complicated but this is still amazing how you got all this to work.  Great job !!